spinalcordtoolbox/moco.py

```diff
@@ -0,0 +1,893 @@
+    file_datasubgroup = 'datasub-groups.nii'  # concatenation of the average of each file_datasub
+    file_bvec = 'bvecs.txt'
+    file_mask = 'mask.nii'
+    file_moco_params_csv = 'moco_params.tsv'
+    file_moco_params_x = 'moco_params_x.nii.gz'
+    file_moco_params_y = 'moco_params_y.nii.gz'
+    ext_data = '.nii.gz'  # workaround "too many open files" by slurping the data
+    # TODO: check if .nii can be used
+    mat_final = 'mat_final/'
+    # ext_mat = 'Warp.nii.gz'  # warping field
+
+    # Start timer
+    start_time = time.time()
+
+    sct.printv('\nInput parameters:', param.verbose)
+    sct.printv('  Input file ............ ' + param.fname_data, param.verbose)
+    sct.printv('  Group size ............ {}'.format(param.group_size), param.verbose)
+
+    # Get full path
+    # param.fname_data = os.path.abspath(param.fname_data)
+    # param.fname_bvecs = os.path.abspath(param.fname_bvecs)
+    # if param.fname_bvals != '':
+    #     param.fname_bvals = os.path.abspath(param.fname_bvals)
+
+    # Extract path, file and extension
+    # path_data, file_data, ext_data = sct.extract_fname(param.fname_data)
+    # path_mask, file_mask, ext_mask = sct.extract_fname(param.fname_mask)
+
+    path_tmp = sct.tmp_create(basename="moco", verbose=param.verbose)
+
+    # Copying input data to tmp folder
+    sct.printv('\nCopying input data to tmp folder and convert to nii...', param.verbose)
+    convert(param.fname_data, os.path.join(path_tmp, file_data))
+    if param.is_diffusion:
+        sct.copy(param.fname_bvecs, os.path.join(path_tmp, file_bvec), verbose=param.verbose)
+    if param.fname_mask != '':
+        convert(param.fname_mask, os.path.join(path_tmp, file_mask), verbose=param.verbose)
+        # Update field in param (because used later in another function, and param class will be passed)
+        param.fname_mask = file_mask
+
+    # Build absolute output path and go to tmp folder
+    curdir = os.getcwd()
+    path_out_abs = os.path.abspath(param.path_out)
+    os.chdir(path_tmp)
+
+    # Get dimensions of data
+    sct.printv('\nGet dimensions of data...', param.verbose)
+    im_data = Image(file_data)
+    nx, ny, nz, nt, px, py, pz, pt = im_data.dim
+    sct.printv('  ' + str(nx) + ' x ' + str(ny) + ' x ' + str(nz), param.verbose)
+
+    # Get orientation
+    sct.printv('\nData orientation: ' + im_data.orientation, param.verbose)
+    if im_data.orientation[2] in 'LR':
+        param.is_sagittal = True
+        sct.printv('  Treated as sagittal')
+    elif im_data.orientation[2] in 'IS':
+        param.is_sagittal = False
+        sct.printv('  Treated as axial')
+    else:
+        param.is_sagittal = False
+        sct.printv('WARNING: Orientation seems to be neither axial nor sagittal.')
+
+    # Set suffix of transformation file name (depends on orientation)
+    param.suffix_mat = '0GenericAffine.mat' if param.is_sagittal else 'Warp.nii.gz'
```

Comment:
here 80b1a151ac1cf1a5c17506cdf815b5c16ef92691

Reply:
Sorry, I wasn't quite clear.  It's clear from the code what is going on, but to a user it might be rather surprising why the file type has changed.  SCT already prints what orientation it has detected, so perhaps it could also print that it's using {mat,warp} as a result.

Reply:
here: 79ac3f441ed4a55cb26c1798d5d1184d6916e42b